tst/unit/test_container_iterator.cpp

```diff
@@ -137,3 +138,156 @@ TEST_CASE("Can pull variables from containers based on Metadata", "[ContainerIte
     }
   }
 }
+//Test wrapper to run a function multiple times
+template<typename InitFunc,typename PerfFunc>
+double performance_test_wrapper(const int n_burn, const int n_perf,
+    InitFunc init_func,PerfFunc perf_func){
+
+  //Initialize the timer and test
+  Kokkos::Timer timer;
+  init_func();
+
+  for( int i_run = 0; i_run < n_burn + n_perf; i_run++){
+    if(i_run == n_burn){
+      //Burn in time is over, start timing
+      Kokkos::fence();
+      timer.reset();
+    }
+
+    //Run the function timing performance
+    perf_func();
+  }
+
+  //Time it
+  Kokkos::fence();
+  double perf_time = timer.seconds();
+
+  //FIXME?
+  //Validate results?
+
+  return perf_time;
+
+}
+
+
+TEST_CASE("Container Iterator Performance",
+          "[ContainerIterator][performance]") {
+
+
+  const int N = 16; //Dimensions of blocks
+  const int n_burn = 500; //Num times to burn in before timing
+  const int n_perf = 500; //Num times to run while timing
+
+  //Make a raw ParArray4D for closest to bare metal looping
+  ParArray4D<Real> raw_array("raw_array",10,N,N,N);
+
+  //Make a function for initializing the raw ParArray4D
+  auto init_raw_array = [&]() {
+    par_for("Initialize ", DevSpace(),
+      0, 10-1,
+      0, N-1,
+      0, N-1,
+      0, N-1,
+      KOKKOS_LAMBDA(const int l, const int k, const int j, const int i) {
+        raw_array(l,k,j,i) = static_cast<Real>( (l+1)*(k+1)*(j+1)*(i+1) );
+      });
+  };
+
+  //Test performance iterating over variables (we should aim for this performance)
+  double time_raw_array = performance_test_wrapper( n_burn, n_perf,init_raw_array,
+    [&](){
+      par_for("Raw Array Perf", DevSpace(),
+        0, 10-1,
+        0, N-1,
+        0, N-1,
+        0, N-1,
+        KOKKOS_LAMBDA(const int l, const int k, const int j, const int i) {
+          raw_array(l,k,j,i) *= raw_array(l,k,j,i); //Do something trivial, square each term
+        });
+    });
+
+
+  //Make a container for testing performance
+  Container<Real> container;
+  Metadata m_in({Metadata::Independent});
+  Metadata m_out;
+  std::vector<int> scalar_block_size {N,N,N};
+  std::vector<int> vector_block_size {3,N,N,N};
+
+  // make some variables - 5 in all, 2 3-vectors, total 10 fields
+  container.Add("v0",m_in, scalar_block_size);
+  container.Add("v1",m_in, scalar_block_size);
+  container.Add("v2",m_in, vector_block_size);
+  container.Add("v3",m_in, scalar_block_size);
+  container.Add("v4",m_in, vector_block_size);
+
```

Comment:
You've only added 9 fields.
```suggestion
container.Add("v5",m_in, scalar_block_size);
```